plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java

```diff
@@ -111,6 +133,15 @@ Transaction newCreateOrReplaceTableTransaction(
             String location,
             Map<String, String> properties);
 
+    Transaction newMigrateTableTransaction(
```

Comment:
I'm not sure if it's worth a whole separate interface method for this. WDYT about just adding a new boolean parameter to `newCreateTableTransaction` like `forInPlaceMigration` or `ignoreExistingContent` or something? (And catalogs can skip the validation if the new txn is for a migration.)

But I'm curious to hear what @findepi thinks about these various approaches

Reply:
I added this because I think it is more clear for the interface if it is for a different operation, besides the actions assumed to be taken are different for create table v.s. migrate table right? Create table is supposed to include that empty location check given that we do not allow shared location and the existing behavior of new table creation.